File: benchmarks/pandas/bench_str_extract_groups.py
\"\"\"Benchmark: str.extract on 10k-element string Series\"\"\"
import json, time
import pandas as pd

ROWS = 10_000
WARMUP = 3
ITERATIONS = 10
data = [f"user_{i}_score_{i % 100}" for i in range(ROWS)]
s = pd.Series(data)

for _ in range(WARMUP):
    s.str.extract(r"user_(\d+)_score_(\d+)")

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.str.extract(r"user_(\d+)_score_(\d+)")
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "str_extract_groups", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_str_extract_groups.ts
/**
 * Benchmark: strExtractGroups on 10k-element string Series
 */
import { Series, strExtractGroups } from "../../src/index.js";

const ROWS = 10_000;
const WARMUP = 3;
const ITERATIONS = 10;
const data = Array.from({ length: ROWS }, (_, i) => `user_${i}_score_${i % 100}`);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) strExtractGroups(s, /user_(\d+)_score_(\d+)/);
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) strExtractGroups(s, /user_(\d+)_score_(\d+)/);
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "str_extract_groups",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
